drivers/spi/spi_silabs_eusart.c

```diff
@@ -298,21 +738,28 @@ static DEVICE_API(spi, spi_silabs_eusart_api) = {
 	.release = spi_silabs_eusart_release,
 };
 
-#define SPI_INIT(n)										\
-	PINCTRL_DT_INST_DEFINE(n);								\
-	static struct spi_silabs_eusart_data spi_silabs_eusart_data_##n = {			\
-		SPI_CONTEXT_INIT_LOCK(spi_silabs_eusart_data_##n, ctx),				\
-		SPI_CONTEXT_INIT_SYNC(spi_silabs_eusart_data_##n, ctx),				\
-		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_DRV_INST(n), ctx)};				\
-	static struct spi_silabs_eusart_config spi_silabs_eusart_cfg_##n = {			\
-		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(n),					\
-		.base = (EUSART_TypeDef *)DT_INST_REG_ADDR(n),					\
-		.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),				\
-		.clock_cfg = SILABS_DT_INST_CLOCK_CFG(n),					\
-		.clock_frequency = DT_INST_PROP_OR(n, clock_frequency, 1000000)			\
-	};											\
+#if defined(CONFIG_SPI_ASYNC)
+#define SPI_DMA_DEVICE_INSTANCE .dma_device = (const struct device *)DEVICE_DT_GET_ONE(silabs_ldma),
```

Comment:
DMA device should be gotten from `dmas` property of the instance, not globally using a compatible